### ISSUE TITLE
Slow switching between view modes

### ISSUE THREAD

**Issue by petermjones** [Flipkart/recyclerlistview#164]
Great component, but one stumbling block that I still have is that switching from a grid view (20 thumbnails per screenful - about 60 'views') over to a list view (about one per screen) on the same screen takes 2-5 seconds to complete.  Flipping from list and grid happens much more quickly. My guess is that the delay is getting rid of the old, unusable views via commands sent over the bridge.  I've tried including the view mode (grid/list) as part of the key for the views, but that does not seem to help.  Any ideas?

Reactotron snapshot notes:
onPressItem function sets the state to 'list'
PostWallScreen is the main screen
GridViewComponent renders either a list of grid view, using RecyclerListView
Note the huge delays - running on an iPhone 6s plus.  Visually, we see something change about 3 seconds after tapping on the grid to list button.
![reactotron](https://user-images.githubusercontent.com/1505036/39436970-63d35da4-4c54-11e8-9a6c-44b4ceffc016.jpg)

Another example, this time after scrolling down the grid view a full screens worth, then switching over to list view.  5 or more seconds before the screen actually changes after tapping.
![reactotron-1](https://user-images.githubusercontent.com/1505036/39437557-0eec48ee-4c56-11e8-9d37-4706045783cc.jpg)

**Comment by petermjones** [comment]
I've also noticed that when switching from grid to list view, even though the render row function returns items that are near the size of the screen, RecyclerListView still asks for dozens of cells - ignoring the fact that the cells are much larger.

I hope there is a proper way to deal with this.  If not, I was thinking that I could have two RecyclerListView's on the screen, one of them zero height, and toggle between the two of them. Not an ideal solution.

**Comment by naqvitalha** [comment]
Try out the view change sample that I've added. See if that helps you: https://snack.expo.io/@naqvitalha/rlv-demo
Ideally there shouldn't be any delay. I can look into a sample if you can provide me one.

**Comment by petermjones** [comment]
Thanks for the sample.  My boss hasn't given me permission to share our code.  In our use-case, our grid uses square thumbnails 1/3 the screen width, and the list images use the full width of the screen.  Because of this, they load different images, perhaps 30Kb vs 200Kb.  Placing a reactotron log into the render row function shows that RecyclerListView is building many views using the large image (during the switch), instead of the 2 or 3 that are needed to fill the current screen and a bit beyond.

When I change the RowRender to just draw coloured squares, switching happens quickly.

To resolve this, I currently have three (we actually have grid/mosaic/list view modes) RecyclerListViews, and I set the height within style to zero for the two that aren't currently 'active'.  They share data, but have different LayoutProviders.  Now, with images drawn, switching happens quickly.

Any problems using multiple RecyclerListViews in this manner?

Finally, I've noticed that while recycling of layout views, touch opacities and images is very clean, recycling of text is not as good.  Performance monitor shows the number of views climbing on list view (which has some text - username, date, number of views, etc), but holding pretty steady on the image-only grid view.  Any trick I can use there?

Thanks.
